pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/tg_owt.nix

```
@@ -1,36 +1,61 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, cmake, ninja, yasm
-, libjpeg, openssl, libopus, ffmpeg, alsa-lib, libpulseaudio, protobuf
-, xorg, libXtst
+{ lib, stdenv, fetchFromGitHub, fetchpatch, pkg-config, cmake, ninja, yasm
+, libjpeg, openssl, libopus, ffmpeg, protobuf, openh264, usrsctp, libvpx
+, libX11, libXtst, libXcomposite, libXdamage, libXext, libXrender, libXrandr, libXi
+, glib, abseil-cpp, pipewire
+, Cocoa, AppKit, IOKit, IOSurface, Foundation, AVFoundation, CoreMedia, VideoToolbox
+, CoreGraphics, CoreVideo, OpenGL, Metal, MetalKit, CoreFoundation, ApplicationServices
 }:
 
-let
-  rev = "2d804d2c9c5d05324c8ab22f2e6ff8306521b3c3";
-  sha256 = "0kz0i381iwsgcc3yzsq7njx3gkqja4bb9fsgc24vhg0md540qhyn";
-
-in stdenv.mkDerivation {
+stdenv.mkDerivation {
   pname = "tg_owt";
-  version = "git-${rev}";
+  version = "unstable-2021-12-12";
 
   src = fetchFromGitHub {
     owner = "desktop-app";
     repo = "tg_owt";
-    inherit rev sha256;
+    rev = "429a6869e4a164e0aad2d8657db341d56f9a6a6f";
+    sha256 = "sha256-aBN2V71ffpJ8N223zqmRBVsjcJ9RrB3QnJcNhhXLZ+A=";
     fetchSubmodules = true;
   };
 
+  patches = lib.optionals stdenv.isDarwin [
```

```suggestion
  patches = [
```
otherwise it could bitrot with the next update.